cardano-api/src/Cardano/Api/Shelley.hs

```diff
@@ -1,95 +1,159 @@
 module Cardano.Api.Shelley
-  ( module Cardano.Api.LocalChainSync
-  , module Cardano.Api.MetaData
-  , module Cardano.Api.Protocol
-  , module Cardano.Api.Shelley.Genesis
-  , module Cardano.Api.Shelley.ITN
-  , module Cardano.Api.TextView
-  , module Cardano.Api.TxSubmit
-  , module Cardano.Api.Typed
-  , module Cardano.Api.Protocol.Cardano
-  , module Cardano.Api.Protocol.Shelley
-  , module Cardano.Api.Protocol.Types
-  , module Cardano.Slotting.Slot
-  , module Ouroboros.Consensus.Block
-  , module Ouroboros.Consensus.BlockchainTime
-  , module Ouroboros.Consensus.Cardano
-  , module Ouroboros.Consensus.Ledger.SupportsMempool
-  , module Ouroboros.Consensus.Network.NodeToClient
-  , module Ouroboros.Consensus.Node.NetworkProtocolVersion
-  , module Ouroboros.Consensus.Node.ProtocolInfo
-  , module Ouroboros.Consensus.Node.Run
-  , module Ouroboros.Network.AnchoredFragment
-  , module Ouroboros.Network.Block
-  , module Ouroboros.Network.Magic
-  , module Ouroboros.Network.Mux
-  , module Ouroboros.Network.NodeToClient
-  , module Ouroboros.Network.Point
-  , module Ouroboros.Network.Protocol.ChainSync.Client
-  , module Ouroboros.Network.Protocol.ChainSync.Type
-  , module Ouroboros.Network.Protocol.LocalTxSubmission.Type
-  , module Shelley.Spec.Ledger.Genesis
-  , module Shelley.Spec.Ledger.OCert
+  ( module Cardano.API,
+    -- * Era
+    Shelley,
+    HasTypeProxy(..),
+    AsType(AsShelleyAddress,
+           AsShelleyTxBody,
+           AsShelleyTx,
+           AsShelleyWitness),
+
+    -- * Cryptographic key interface
+    -- $keys
+    VerificationKey(..),
+
+    -- * Payment addresses
+    -- | Constructing and inspecting Shelley payment addresses
+    Address(ShelleyAddress),
+    NetworkId(Mainnet, Testnet),
+
+    -- * Building transactions
+    -- | Constructing and inspecting transactions
+    TxBody(ShelleyTxBody),
+    TxId(TxId),
+    TxIn(TxIn),
+    TxOut(TxOut),
+    TxIx(TxIx),
+    Lovelace(Lovelace),
+    SlotNo(SlotNo),
+
+    -- * Signing transactions
+    -- | Creating transaction witnesses one by one, or all in one go.
+    Tx(ShelleyTx),
+
+    -- ** Incremental signing and separate witnesses
+    Witness (ShelleyBootstrapWitness, ShelleyKeyWitness, ShelleyScriptWitness),
+    ShelleyWitnessSigningKey (WitnessPaymentKey,
+                              WitnessPaymentExtendedKey,
+                              WitnessStakeKey,
+                              WitnessStakeExtendedKey,
+                              WitnessStakePoolKey,
+                              WitnessGenesisKey,
+                              WitnessGenesisExtendedKey,
+                              WitnessGenesisDelegateKey,
+                              WitnessGenesisDelegateExtendedKey),
```

Comment:
Overall, I don't think we should be aligning our code in this way and I'll just give a couple of reasons:

1. If the name `ShelleyWitnessSigningKey` changes, then all of these lines need to be realigned (adds `diff` noise and is error prone in the sense that some may not notice that they need to re-align if they're doing a project-wide search-and-replace, for example)

2. I'd say that, generally, we should avoid variable-length indentation. I think the `cardano-wallet` wiki makes a good point on this: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#avoid-variable-length-indentation. Although, I do acknowledge that we already have a bunch of variable-length indentation in the codebase.

3. Also, generally, I think we should stick with the "comma-leading" style:
```
    ShelleyWitnessSigningKey 
      ( WitnessPaymentKey
      , WitnessPaymentExtendedKey
      , WitnessStakeKey
      ...
      ),
```

Reply:
Also, side note, in the case above, aren't those just normal constructor exports? So couldn't you just export `ShelleyWitnessSigningKey (..)`? Or are you doing it this way for added readability?

If so, the trouble with that comes when a new constructor is added and one may forget to export it.

Reply:
As a further points to the ones Luke has already made, align should aim to be invariant with other changes.

Eg:
```
import A      (A (..), toA)
import B      (B (..), toB)
```
will require re-indenting these import lists as soon as we import say `Cardano.DbSync.Era.Shelley.Insert`.

This means the following is better
```
import A (A (..), toA)
import B (B (..), toB)
```
because the alignment rule would not require any changes to the existing imports if `Cardano.DbSync.Era.Shelley.Insert` were added.

This rule can be boiled down to "use indent rules that as maximally resistant to requiring semantically null changes".

I also prefer not to needlessly consume vertical code space (from working too long on laptops no doubt) so I would probably format that as:
```
    ShelleyWitnessSigningKey (WitnessPaymentKey, WitnessPaymentExtendedKey, WitnessStakeKey,
        WitnessStakeExtendedKey, WitnessStakePoolKey, WitnessGenesisKey, WitnessGenesisExtendedKey,
        WitnessGenesisDelegateKey, WitnessGenesisDelegateExtendedKey),
```

This reminds me of another formatting rule I like is that indentation should suggest nesting. Once thing I see all the time in out code base which I dislike is:
```
wibble x = case x of
   A -> ....
   B -> ....
```
which afaiac is far less clear than:
```
wibble x =
  case x of
     A -> ....
     B -> ....
 ```
where the `A` and `B` are much more obviously heirarchically subserviant to the `case`.

Reply:
> 

Also, side note, in the case above, aren't those just normal constructor exports? So couldn't you just export ShelleyWitnessSigningKey (..)? Or are you doing it this way for added readability?

If so, the trouble with that comes when a new constructor is added and one may forget to export it.

I went for an explicit approach because it makes it clearer which era the constructors belong to. If we do decide to export more upstream library modules through `cardano-api` this would be preferable as you could instantly figure out where constructors/types are coming, saving us the usual wild goose chase. I think this is the better trade off.

Reply:
>Also, side note, in the case above, aren't those just normal constructor exports? So couldn't you just export ShelleyWitnessSigningKey (..)? Or are you doing it this way for added readability?

>If so, the trouble with that comes when a new constructor is added and one may forget to expo

 When I was under the impression that we were also exporting upstream library modules via `cardano-api`, I explicitly exported constructors to avoid the usual wild goose chase when trying to figure out what came from which module. I still think its of value at the potential expense of forgetting to export a constructor.